packages/nc-gui/components/smartsheet/Grid.vue

```diff
@@ -114,16 +116,57 @@ const { selectCell, selectBlock, selectedRange, clearRangeRows, startSelectRange
   isPkAvail,
   clearCell,
   makeEditable,
-  () => {
-    if (selected.row !== null && selected.col !== null) {
+  (row?: number | null, col?: number | null) => {
+    row = row ?? selected.row
+    col = col ?? selected.col
+    if (row !== undefined && col !== undefined && row !== null && col !== null) {
       // get active cell
-      const td = tbodyEl.value?.querySelectorAll('tr')[selected.row]?.querySelectorAll('td')[selected.col + 1]
-      if (!td) return
+      const rows = tbodyEl.value?.querySelectorAll('tr')
+      const td = rows?.[row].querySelectorAll('td')[col === 0 ? 0 : col + 1]
+
+      if (!td || !gridWrapper.value) return
+
+      const { height: headerHeight } = tableHead.value!.getBoundingClientRect()
+
+      const childPos = td.getBoundingClientRect()
+      const parentPos = gridWrapper.value.getBoundingClientRect()
+      const relativePos = {
+        top: childPos.top - parentPos.top,
+        right: childPos.right - parentPos.right,
+        bottom: childPos.bottom - parentPos.bottom,
+        left: childPos.left - parentPos.left,
+      }
+
+      if (rows && row === rows.length - 2) {
+        // if last row make 'Add New Row' visible
+        gridWrapper.value.scrollTo({
+          top: gridWrapper.value.scrollHeight,
+          left:
+            relativePos.right > 0
+              ? gridWrapper.value.scrollLeft + relativePos.right + 9 // 9 is for border
+              : relativePos.left < 0
+              ? gridWrapper.value.scrollLeft + relativePos.left
+              : gridWrapper.value.scrollLeft,
+          behavior: 'smooth',
+        })
+        return
+      }
+
       // scroll into the active cell
-      td.scrollIntoView({
+      gridWrapper.value.scrollTo({
+        top:
+          relativePos.bottom > 0
+            ? gridWrapper.value.scrollTop + relativePos.bottom + 9 // 9 is for border
+            : relativePos.top - headerHeight < 0
+            ? gridWrapper.value.scrollTop + relativePos.top - headerHeight
+            : gridWrapper.value.scrollTop,
+        left:
+          relativePos.right > 0
+            ? gridWrapper.value.scrollLeft + relativePos.right + 9 // 9 is for border
+            : relativePos.left < 0
+            ? gridWrapper.value.scrollLeft + relativePos.left
+            : gridWrapper.value.scrollLeft,
```

Comment:
It would be nice if these sort of blocks had some explaining comments so other users/contributors understand what it exactly does.
Just by looking at it, it would be hard to discern what these blocks actually calculate 😄

Reply:
Might be even better so outsource the logic into a function instead of inlining these conditions :)

Reply:
Implemented `getContainerScrollForElement` function, tried to comment as well (mostly geometrical calculations so don't know how to better comment for them)